scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -209,7 +209,9 @@ def discover(self):
         self.test_config = TestConfiguration(self.env.test_config)
 
         self.add_configurations()
-        num = self.add_testsuites(testsuite_filter=self.run_individual_testsuite)
+        num = self.add_testsuites(testsuite_filter=self.run_individual_testsuite,
+                                testsuite_pattern=self.options.test_pattern)
```

Comment:
```suggestion
        num = self.add_testsuites(testsuite_filter=self.options.test,
                                testsuite_pattern=self.options.test_pattern)
```
use self.options.test instead of self.run_individual_testsuite,
and remove lines:
```
        if self.options.test:
            self.run_individual_testsuite = self.options.test
```
to not filter in out in apply_filters method